src/visitor.js

```diff
@@ -30,45 +29,42 @@ var Visitor = function(options) {
 };
 
 Visitor.loadVisitor = function(visitorId) {
-  var deferred = $.Deferred(),
-    resolve = function(attrs) {
-      deferred.resolve(new Visitor(attrs));
-    };
-
   if (visitorId) {
     if (TestTrackConfig.getAssignments()) {
-      resolve({
-        id: visitorId,
-        assignments: TestTrackConfig.getAssignments(),
-        ttOffline: false
-      });
+      return Promise.resolve(
+        new Visitor({
+          id: visitorId,
+          assignments: TestTrackConfig.getAssignments(),
+          ttOffline: false
+        })
+      );
     } else {
-      client
+      return client
         .get('/v1/visitors/' + visitorId, { timeout: 5000 })
-        .then(attrs => {
-          resolve({
-            id: attrs['id'],
-            assignments: Assignment.fromJsonArray(attrs['assignments']),
+        .then(({ data }) => {
+          return new Visitor({
+            id: data['id'],
```

Comment:
is there a reason we are using bracket notation instead of dot access?
```suggestion
            id: data.id,
```

Reply:
Not sure why it was originally, but happy to change it to dot access.